homeassistant/components/alarm_control_panel/device_trigger.py

```diff
@@ -129,14 +138,17 @@ async def async_attach_trigger(
     elif config[CONF_TYPE] == "disarmed":
         from_state = STATE_ALARM_TRIGGERED
```

Comment:
@balloob Schould the "disarmed" trigger not fire always when the alarm is disarmed, also if the from_state is one of the armed states instead of alarm_triggered?
Schould I just remove the from_state?

Reply:
Yes.